src/app/screens/Home/index.js

```diff
@@ -1,18 +1,56 @@
 import React, {useState} from 'react';
-import {View, TextInput} from 'react-native';
+import {
+  View,
+  TextInput,
+  Text,
+  TouchableOpacity,
+  ImageBackground,
+} from 'react-native';
 import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
+import {useDispatch} from 'react-redux';
 
 import styles from './styles';
 import Button from './components/Button';
 import {retrieveButtons, execOpKeyboardKeyPresed} from './utils';
 
-const Home = () => {
+import actionHistory from '../../redux/history/actions';
+
+const Home = ({navigation}) => {
   const [display, setDisplay] = useState('');
-  const buttons = retrieveButtons(display, setDisplay);
+  const [saveExpression, setSaveExpression] = useState('');
+  const buttons = retrieveButtons(display, setDisplay, setSaveExpression);
+  const dispatch = useDispatch();
 
   return (
     <KeyboardAwareScrollView style={styles.contain}>
       <View style={styles.containerCalculator}>
+        <View style={styles.boxHistory}>
+          {saveExpression.length > 0 && (
+            <>
+              <Text style={styles.textValues}>
+                {saveExpression.split(/[=]/)[0]}
+              </Text>
+              <Text style={styles.textIqual}>=</Text>
+              <Text style={styles.textValues}>
+                {saveExpression.split(/[=]/)[1]}
+              </Text>
+              <View style={styles.boxButtonHistory}>
+                <TouchableOpacity
+                  onPress={() =>
```

Comment:
extract dispatch into a function, also, you can disable the touchable id SaveExpression.length<=0

Reply:
I already included this in utils, and disable the button